yellowbrick/classifier.py

```diff
@@ -50,6 +51,149 @@ def __init__(self, model, ax=None, **kwargs):
 
         super(ClassificationScoreVisualizer, self).__init__(model, ax=ax, **kwargs)
 
+##########################################################################
+## ConfusionMatrix
+##########################################################################
+
+class ConfusionMatrix(ClassificationScoreVisualizer):
+    """
+    Creates a heatmap visualization of the sklearn.metrics.confusion_matrix().
+    Initialization: Requires a classification model
+    """
+    def __init__(self, model, ax=None, classes=None, **kwargs):
+        """
+        Provide a classifier model
+
+        Parameters
+        ----------
+        :param model: the Scikit-Learn estimator
+            Should be an instance of a classifier, else the __init__ will
+            return an error.
+
+        :param ax: the matplotlib axis to plot the figure on (if None, a new axis will be created)
+
+        :param classes: a list of class names to use in the confusion_matrix.
+            This is passed to the 'labels' parameter of sklearn.metrics.confusion_matrix(), and follows the behaviour
+            indicated by that function. It may be used to reorder or select a subset of labels.
+            If None, values that appear at least once in y_true or y_pred are used in sorted order.
+            Default: None
+        """
+        super(ConfusionMatrix, self).__init__(model, ax=ax, **kwargs)
+        #Parameters provided by super (for reference during development only):
+        #self.ax
+        #self.size
+        #self.color
+        #self.title
+        #self.estimator
+        #self.name
+
+        #Initialize all the other attributes we'll use (for coder clarity)
+        self.confusion_matrix = None
+
+        self.cmap = color_sequence(kwargs.pop('cmap', 'YlOrRd'))
+
+        #If possible, assign the classes. If this can't happen now, it will happen in the .fit() method
```

Comment:
So `_classes` is a Python thing, and signals to developers that this is an internal property and shouldn't be messed with, whereas `classes` is the external property (there is no public, private, or protected in Python). Note that `__classes` causes name mangling. 

http://radek.io/2011/07/21/private-protected-and-public-in-python/

The name `classes_` is actually a Scikit-Learn thing, from the [docs](http://scikit-learn.org/stable/tutorial/statistical_inference/settings.html)

> Estimated parameters: When data is fitted with an estimator, parameters are estimated from the data at hand. All the estimated parameters are attributes of the estimator object ending by an underscore: